gdal/ogr/ogrsf_frmts/ngw/ngw_api.cpp

```diff
@@ -0,0 +1,643 @@
+    }
+}
+
+Permissions CheckPermissions(const std::string &osUrl,
+    const std::string &osResourceId, char **papszHTTPOptions, bool bReadWrite)
+{
+    Permissions stOut;
+    CPLErrorReset();
+    CPLJSONDocument oPermissionReq;
+    bool bResult = oPermissionReq.LoadUrl( GetPermisions( osUrl, osResourceId ),
+        papszHTTPOptions );
+
+    CPLJSONObject oRoot = oPermissionReq.GetRoot();
+    if( oRoot.IsValid() )
+    {
+        if( bResult )
+        {
+            stOut.bResourceCanRead = oRoot.GetBool( "resource/read", true );
+            stOut.bResourceCanCreate = oRoot.GetBool( "resource/create", bReadWrite );
+            stOut.bResourceCanUpdate = oRoot.GetBool( "resource/update", bReadWrite );
+            stOut.bResourceCanDelete = oRoot.GetBool( "resource/delete", bReadWrite );
+
+            stOut.bDatastructCanRead = oRoot.GetBool( "datastruct/read", true );
+            stOut.bDatastructCanWrite = oRoot.GetBool( "datastruct/write", bReadWrite );
+
+            stOut.bDataCanRead = oRoot.GetBool( "data/read", true );
+            stOut.bDataCanWrite = oRoot.GetBool( "data/write", bReadWrite );
+
+            stOut.bMetadataCanRead = oRoot.GetBool( "metadata/read", true );
+            stOut.bMetadataCanWrite = oRoot.GetBool( "metadata/write", bReadWrite );
+        }
+        else
+        {
+            std::string osErrorMessage = oRoot.GetString("message");
+            if( osErrorMessage.empty() )
+            {
+                osErrorMessage = "Get permissions failed";
+            }
+            CPLErrorSetState(CE_Failure, CPLE_AppDefined, osErrorMessage.c_str());
+        }
+    }
+    else
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Get permissions failed");
+    }
+
+    return stOut;
+}
+
+std::string GetFeatureCount(const std::string &osUrl, const std::string &osResourceId)
+{
+    return osUrl + "/api/resource/" + osResourceId + "/feature_count";
+}
+
+std::string GetLayerExtent(const std::string &osUrl, const std::string &osResourceId)
+{
+    return osUrl + "/api/resource/" + osResourceId + "/extent";
+}
+
+std::string GetResmetaSuffix(CPLJSONObject::Type eType)
+{
+    switch( eType ) {
+        case CPLJSONObject::Integer:
+        case CPLJSONObject::Long:
+            return ".d";
+        case CPLJSONObject::Double:
+            return ".f";
+        default:
+            return "";
+    }
+}
+
+void FillResmeta(CPLJSONObject &oRoot, char **papszMetadata)
+{
+    CPLJSONObject oResMeta("resmeta", oRoot);
+    CPLJSONObject oResMetaItems("items", oResMeta);
+    CPLStringList oaMetadata(papszMetadata, FALSE);
+    for( int i = 0; i < oaMetadata.size(); ++i )
+    {
+        std::string osItem = oaMetadata[i];
+        size_t nPos = osItem.find("=");
+        if( nPos != std::string::npos )
+        {
+            std::string osItemName = osItem.substr( 0, nPos );
+            CPLString osItemValue = osItem.substr( nPos + 1 );
+
+            size_t nSuffixPos = osItemName.size() - 2;
```

Comment:
check that osItemName.size() > 2

Reply:
Fix per 4579aeab927d70ddab6142340bf5550c6789b42e